File: getStocks.js
const Alpaca = require('@alpacahq/alpaca-trade-api')
const express = require('express')
const bodyParser = require('body-parser');

const alpaca = new Alpaca({
	keyId: "PKRZGWVR3MNK33BT8R4A",
	secretKey:"gqkhQtP5/OSd1iOqFc8qvbSN8HJmftXhDS1IWap5",
	paper: true,
	usePolygon: false
})


function getStockPrices(symbolList) {
	const barset = alpaca.getBars(
		'1Min',
		symbolList,
		{	
			limit: 1000,
			start:'2020-05-12T09:30:00-04:00',
			end: '2020-05-1209:30:00-04:00',
		}
	)

	return barset
}


const app = express();
app.use(bodyParser.json());
app.use(express.static('public'))
app.get('/watchlist', (req, res) => getStockPrices(req.query.symbols).then((barset) => {
								res.send(barset);
							})
		)

app.listen(3000, () => {
	console.log("Listening on port 3000");
	
})

File: get_stocks.py
import requests, json
import alpaca_trade_api as tradeapi
import pandas as pd
from config import *


BASE_URL = "https://paper-api.alpaca.markets"
DATA_URL = "https://data.alpaca.markets"

api = tradeapi.REST(key_id=API_KEY, secret_key=SECRET_KEY, base_url=BASE_URL, api_version='v2')


#.get_barset(symbols, timeframe, limit, start=None, end=None, after=None, until=None)
#symbols --> String, or list of strings
#timeframe --> minute, 1Min, 5Min, 15Min, day,

#return: t: beginning time, o: open price, h: high price, l: low price, c: close price
#		 v: volume

time = pd.Timestamp(year=2020, month=5, day=8, hour=9)
barset = api.get_barset(symbols="AAPL", timeframe="minute", start=time, limit=20)



def to_pandas(barset):
	dataframe = {}
	for symbol in barset.keys():
		bars = barset[symbol]

		data = {"close": [bar.c for bar in bars],
				"high": [bar.h for bar in bars],
				"low": [bar.l for bar in bars],
				"open": [bar.o for bar in bars],
				"time": [bar.t for bar in bars],
				"volume": [bar.v for bar in bars]}
		dataframe[symbol] = pd.DataFrame(data)
	return dataframe


print (to_pandas(barset))